src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/AccountPicker.cs

```diff
@@ -224,70 +226,68 @@ private async Task<WebAccountProvider> ShowPicker_UWPAsync()
 
         private async void Authenticator_AccountCommandsRequested(
             AccountsSettingsPane sender,
-            AccountsSettingsPaneCommandsRequestedEventArgs e)
+            AccountsSettingsPaneCommandsRequestedEventArgs args)
         {
             AccountsSettingsPaneEventDeferral deferral = null;
             try
             {
-                deferral = e.GetDeferral();
+                deferral = args.GetDeferral();
 
                 if (string.Equals("common", _authority.TenantId))
                 {
                     _logger.Verbose("Displaying selector for common");
-                    e.WebAccountProviderCommands.Add(
-                        new WebAccountProviderCommand(
-                            await WebAuthenticationCoreManager.FindAccountProviderAsync("https://login.microsoft.com", "consumers"),
-                            WebAccountProviderCommandInvoked));
-
-                    e.WebAccountProviderCommands.Add(
-                        new WebAccountProviderCommand(
-                            await WebAuthenticationCoreManager.FindAccountProviderAsync("https://login.microsoft.com", "organizations"),
-                            WebAccountProviderCommandInvoked));
+                    await AddSelectorsAsync(
+                        args, 
+                        addOrgAccounts: true, 
+                        addMsaAccounts: true).ConfigureAwait(true);
                 }
                 else if (string.Equals("organizations", _authority.TenantId))
                 {
                     _logger.Verbose("Displaying selector for organizations");
-
-                    e.WebAccountProviderCommands.Add(
-                       new WebAccountProviderCommand(
-                           await WebAuthenticationCoreManager.FindAccountProviderAsync("https://login.microsoft.com", "organizations"),
-                           WebAccountProviderCommandInvoked));
+                    await AddSelectorsAsync(
+                        args, 
+                        addOrgAccounts: true, 
+                        addMsaAccounts: _isMsaPassthrough).ConfigureAwait(true);
```

Comment:
This is the reason we need apps to tell MSAL if they are MSA-PT or not. MSAL fully controls the Account Picker control by saying "add AAD accounts" and "add MSA accounts". For 3rd party apps, we can rely on the tenant to do this. For 1st party MSA-PT, we can't.

Reply:
aren't 1st party supposed to use /organizations when doing MSA-PT? how do we detect this for 3rd party apps? do you have a doc you can share on MSA-PT? would help w/reviewing.

---
In reply to: [564740521](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2365#discussion_r564740521) [](ancestors = 564740521)

Reply:
Ok..i understand this better now, i've been doing some research. 

---
In reply to: [564868646](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2365#discussion_r564868646) [](ancestors = 564868646,564740521)

Reply:
For browser based auth, ESTS takes care of this because they control the contents of the browser. For WAM-based auth, MSAL controls the contents of the account picker, so we need to know. 


|   | normal app  | msa-pt app  | 
|---|---|---|
| /common  | MSA + AAD  | not supported, server error | 
| /consumers  | MSA  | not supported, server error  |  
| /organizations  | AAD  | MSA+ AAD  | 
| tenanted | AAD  | MSA+ AAD  |